### PR TITLE
Convert Jetty 8.0 groovy to java

### PR DESCRIPTION
**Pull request by heyams**
related to https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7195

### CODE REVIEW

**Comment by heyams** [comment]
does anyone know how to resolve 'Unable to load class 'Io_opentelemetry_instrumentation_muzzleGenerationPlugin'."? 
i followed [troubleshooting guide](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/intellij-setup-and-troubleshooting.md), same outcome though.

**Comment by mateuszrzeszutek** [comment]
> does anyone know how to resolve 'Unable to load class 'Io_opentelemetry_instrumentation_muzzleGenerationPlugin'."? i followed [troubleshooting guide](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/intellij-setup-and-troubleshooting.md), same outcome though.

When does that happen?
Just to make sure -- in the last few days we've updated gradle version to 8.0, can you merge/rebase with `main` and try again?

**Comment by heyams** [comment]
> > does anyone know how to resolve 'Unable to load class 'Io_opentelemetry_instrumentation_muzzleGenerationPlugin'."? i followed [troubleshooting guide](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/intellij-setup-and-troubleshooting.md), same outcome though.
> 
> When does that happen? Just to make sure -- in the last few days we've updated gradle version to 8.0, can you merge/rebase with `main` and try again?

It was auto resolved after I reboot my machine. Somehow my IntelliJ was in a weird state and messed up gradle. Thanks.

File: instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    options.setHttpAttributes(
+        unused ->
+            Sets.difference(
+                DEFAULT_HTTP_ATTRIBUTES, Collections.singleton(SemanticAttributes.HTTP_ROUTE)));
+    options.setHasResponseSpan(endpoint -> endpoint == REDIRECT || endpoint == ERROR);
+    options.setExpectedException(new IllegalStateException(EXCEPTION.getBody()));
+    super.configure(options);
```
Comment (mateuszrzeszutek) [inline instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java:90]: Nit: `super.configure()` does nothing, it was added just for extension.
```suggestion
```

File: instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        () -> {
+          try {
+            return extracted(request, response, endpoint);
+          } catch (IOException e) {
+            throw new RuntimeException(e);
+          }
+        });
+  }
+
+  private static HttpServletResponse extracted(
```
Comment (mateuszrzeszutek) [inline instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java:118]: `extracted()`?

File: instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+  private static class TestHandler extends AbstractHandler {
+    @Override
+    public void handle(
+        String target,
+        Request baseRequest,
+        HttpServletRequest request,
+        HttpServletResponse response)
+        throws IOException, ServletException {
+      if (baseRequest.getDispatcherType() != DispatcherType.ERROR) {
```
Comment (mateuszrzeszutek) [inline instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java:164]: Can you add https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/1649a90e29cb5fad7924a9896a21be5ef525c4a6/instrumentation/jetty/jetty-8.0/javaagent/src/test/groovy/JettyHandlerTest.groovy#L133-L135 back? We've had instrumentations before that decorated the `HttpServletResponse`, even though Jetty internally depends on the fact that it's actually supposed to be `Response`

**Review by mateuszrzeszutek — COMMENTED** [review]
Thanks for the help @heyams !

File: instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java (hunk 3 above)
Comment (mateuszrzeszutek) [reply instrumentation/jetty/jetty-8.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/v8_0/JettyHandlerTest.java:163]: It was fixed -- I think we should leave that one cast/assertion as a regression test though.

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by laurit — APPROVED** [review]

**Pull request merged**